state/application.go

```diff
@@ -268,11 +268,18 @@ func (op *DestroyApplicationOperation) Build(attempt int) ([]txn.Op, error) {
 	case errAlreadyDying:
 		return nil, jujutxn.ErrNoOperations
 	case nil:
-		return ops, nil
-	}
-	if op.Force {
-		logger.Warningf("forcing destroy application for %v despite error %v", op.app.Name(), err)
-		return ops, nil
+		nErrors := len(op.Errors)
+		if nErrors == 0 {
+			return ops, nil
+		}
+		logger.Debugf("%d error%s encountered during %s removal", nErrors, plural(nErrors), op.app.Name())
```

Comment:
I am pretty sure that you don't need to log these errors here because we log them after the Operation has been executed. This also means that you do need nErrors.

Reply:
That's a great point. Have removed. Simplifies things a lot.